### PR TITLE
Calculate table size of FastLRUCache more accurately

### PR DESCRIPTION
**Pull request by guidotag**
Summary: Calculate the required size of the hash table in FastLRUCache more accurately.

Test plan: ``make -j24 check``

### CODE REVIEW

File: cache/lru_cache_test.cc [hunk 1]
```diff
@@ -238,6 +238,26 @@ class FastLRUCacheTest : public testing::Test {
 
   Status Insert(char key, size_t len) { return Insert(std::string(len, key)); }
 
+  uint8_t CalcHashBitsWrapper(size_t capacity, size_t estimated_value_size, CacheMetadataChargePolicy metadata_charge_policy) {
+    return fast_lru_cache::LRUCacheShard::CalcHashBits(capacity, estimated_value_size, metadata_charge_policy);
+  }
+
+  size_t CalcHandleCharge(size_t estimated_value_size, CacheMetadataChargePolicy metadata_charge_policy) {
+    LRUHandle h;
+    h.CalcTotalCharge(estimated_value_size, kDontChargeCacheMetadata);
```
Comment (gitbw95) [inline cache/lru_cache_test.cc]: pls use kDefaultCacheMetadataChargePolicy

File: cache/lru_cache_test.cc [hunk 2]
```diff
@@ -253,6 +273,48 @@ TEST_F(FastLRUCacheTest, ValidateKeySize) {
   EXPECT_NOK(Insert('f', 0));
 }
 
+TEST_F(FastLRUCacheTest, CalcHashBitsTest) {
+  size_t capacity = 1024;
+  size_t estimated_value_size = 1;
+  CacheMetadataChargePolicy metadata_charge_policy = kDontChargeCacheMetadata;
```
Comment (gitbw95) [inline cache/lru_cache_test.cc:279]: pls use kDefaultCacheMetadataChargePolicy

File: cache/fast_lru_cache.h [hunk 3]
```diff
@@ -365,6 +365,8 @@ class ALIGN_AS(CACHE_LINE_SIZE) LRUCacheShard final : public CacheShard {
 
  private:
   friend class LRUCache;
+  friend class FastLRUCacheTest;
```
Comment (gitbw95) [inline cache/fast_lru_cache.h:368]: Try 
ROCKSDB_NAMESPACE::FastLRUCacheTest

File: cache/fast_lru_cache.h [hunk 4]
```diff
@@ -365,6 +368,8 @@ class ALIGN_AS(CACHE_LINE_SIZE) LRUCacheShard final : public CacheShard {
 
  private:
   friend class LRUCache;
+  friend class ROCKSDB_NAMESPACE::fast_lru_cache::FastLRUCacheTest;
```
Comment (gitbw95) [inline cache/fast_lru_cache.h:371]: ROCKSDB_NAMESPACE::fast_lru_cache:: should not be needed here.

Comment (guidotag) [reply cache/fast_lru_cache.h:371]: You are right.

**Review by gitbw95 — APPROVED** [review]
LGTM!

**Pull request closed without merging**